src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2930,6 +2931,11 @@ uct_ib_mlx5_devx_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
         } else if (status != UCS_ERR_UNSUPPORTED) {
             return status;
         }
+        atomic_rkey = memh->atomic_rkey;
+    } else if (uct_ib_mlx5_devx_memh_has_ro(md, memh)) {
+        atomic_rkey = memh->mrs[UCT_IB_MR_STRICT_ORDER].super.ib->rkey;
```

Comment:
maybe we should init memh->atomic_rkey to this value if we need atomic access but it is disabled or failed? somehow combine it with the condition in lines 2918-2925